pkg/dogstatsd/mapper/mapper.go

```diff
@@ -0,0 +1,124 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// Mapping feature is inspired by https://github.com/prometheus/statsd_exporter
+
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	allowedWildcardMatchPattern = regexp.MustCompile(`^[a-zA-Z0-9\-_*.]+$`)
+)
+
+const (
+	matchTypeWildcard = "wildcard"
+	matchTypeRegex    = "regex"
+)
+
+// MetricMapper contains mappings and cache instance
+type MetricMapper struct {
+	Mappings []MetricMapping
+	cache    *mapperCache
+}
+
+// MetricMapping represent one mapping rule
+type MetricMapping struct {
```

Comment:
As said in another comment IMO this should be moved to the config package. We should also map `MatchType` string to an enum there.

Reply:
Sure, will move it to config package once the format is decided: https://github.com/DataDog/datadog-agent/pull/4471#discussion_r351188151

Reply:
> As said in another comment IMO this should be moved to the config package.

The mapping configuration is quite specific to the mapper, so I think it's easier to maintain if it stays in the mapper package.

That's also what's done for `trace` https://github.com/DataDog/datadog-agent/blob/18882ff18356f074dd86923bdb508efb7f2c7be1/pkg/trace/config/apply.go

and logs https://github.com/DataDog/datadog-agent/blob/8979cb6aeec7e7e09d4c0b6dfa830936d380a13b/pkg/logs/config/endpoints.go

Maybe I can move those config objects in a config folder inside `dogstatsd`.

> We should also map MatchType string to an enum there.

I'm not sure it worth it here, since the MatchType is processed only once in `NewMetricMapper`

Reply:
Let's discuss on the other thread